core/shared/src/main/scala/com/softwaremill/sttp/SttpBackendOptions.scala

```diff
@@ -56,18 +74,21 @@ object SttpBackendOptions {
     Empty.socksProxy(host, port)
 
   private def loadSystemProxy: Option[Proxy] = {
-    def system(hostProp: String, portProp: String, make: (String, Int) => Proxy, defaultPort: Int) = {
+    def system(hostProp: String, portProp: String, nonProxyHostsPropOption: Option[String], make: (String, Int, List[String]) => Proxy, defaultPort: Int) = {
       val host = Option(System.getProperty(hostProp))
       def port = Try(System.getProperty(portProp).toInt).getOrElse(defaultPort)
-      host.map(make(_, port))
+      def nonProxyHosts: List[String] = nonProxyHostsPropOption
+        .flatMap(nonProxyHostsProp => Try(System.getProperty(nonProxyHostsProp).split("|").toList).toOption)
+        .getOrElse(Nil)
+      host.map(make(_, port, nonProxyHosts))
     }
 
-    def proxy(t: ProxyType)(host: String, port: Int) = Proxy(host, port, t)
+    def proxy(t: ProxyType)(host: String, port: Int, nonProxyHosts: List[String]) = Proxy(host, port, t, nonProxyHosts)
 
     import ProxyType._
-    val socks = system("socksProxyHost", "socksProxyPort", proxy(Socks), 1080)
-    val http = system("http.proxyHost", "http.proxyPort", proxy(Http), 80)
-    val https = system("https.proxyHost", "https.proxyPort", proxy(Http), 443)
+    val socks = system("socksProxyHost", "socksProxyPort", None, proxy(Socks), 1080)
+    val http = system("http.proxyHost", "http.proxyPort", Some("http.nonProxyHosts"), proxy(Http), 80)
+    val https = system("https.proxyHost", "https.proxyPort", Some("http.nonProxyHosts"), proxy(Http), 443)
```

Comment:
shouldn't this be `https.nonProxyHosts`?

Reply:
I'm following this spec : https://docs.oracle.com/javase/8/docs/api/java/net/doc-files/net-properties.html And it says https would use the nonProxyHosts of http